### ISSUE TITLE
pasting is very unperformant in CIDER

### ISSUE THREAD

**Issue by ara4n** [vector-im/riot-web#10687]
i just tried to see how fast i could paste 🎉 with slate, and it was pretty stable other than occasional GCs. On CIDER i can only manage a few per second, and it constantly GCs

**Comment by bwindels** [comment]
Huh... this interesting. I can reproduce... sort off.

I can only reproduce in Chrome when the editor is quite narrow. When profiling this, the relayout takes~ 30ms and the CIDER code only ~1ms. I assume this is caused by the editor needing to grow in height frequently, and a relayout with the document as root is done for every paste. No GCs during any of this, although the javascript vm heap is only ~55mb while trying this, so maybe there is less memory pressure to GC.

I also get this while using Slate btw, so I'm likely seeing something else than you are.

![paste-tada](https://user-images.githubusercontent.com/274386/63936098-8b9b9580-ca4e-11e9-8741-b1c376bfdd9e.gif)

Could you perhaps pass me your profiling session file to compare?

**Comment by ara4n** [comment]
https://matrix.org/_matrix/media/r0/download/matrix.org/dfSBwcLpjwfNqwzbsGHTutzr

**Comment by ara4n** [comment]
btw, i don't think this is a very big deal - it's just that i was trying to benchmark it relative to Slate by pasting a lot of stuff, and got confused when it was ~20x slower to paste.

**Comment by bwindels** [comment]
Ok, the provided recording doesn't clear up much :( I only see one input event being processed by CIDER, at the 7.288s mark, taking just around ~3ms to complete. Lot's of react stack traces and worker messages in there as well, but hard to tell if related to the slowness. Maybe something else causing GCs is preventing the paste events from being dispatched? Firefox profiler sparseness also doesn't help here. If you can repro in Chrome, a recording from that browser could help.

**Comment by jryans** [comment]
(The default Firefox profiler is near-useless, but a better one exists as an option in Firefox Nightly... The new profiling UI is a complete rewrite and loads the profile in a browser tab among many other features. I recommend it where possible as the profile viewer is much fancier, but it does involve using Nightly and checking "Enable new performance recorder" in DevTools settings. Unfortunately, I believe it uses a separate profile format, so we'd need a fresh profile to be recorded in the new tool to analyse it there.)

**Comment by bwindels** [comment]
> (The default Firefox profiler is near-useless, but a better one exists as an option in Firefox Nightly... 

Indeed, must be a different format. Tried loading the file in nightly and gave me an identical result.

**Comment by jryans** [comment]
Actually an even easier way to access the new Firefox profiler is via about:config setting "devtools.performance.new-panel-enabled" to true, as this will work in any release channel. Then go to DevTools and record in the performance panel.

**Comment by t3chguy** [comment]
Closing this as presumed fixed in the many years due to lack of further reported cases
